scipy/io/wavfile.py

```diff
@@ -421,25 +425,53 @@ def _read_data_chunk(fid, format_tag, channels, bit_depth, is_big_endian,
     # Number of bytes per sample (sample container size)
     bytes_per_sample = block_align // channels
     n_samples = size // bytes_per_sample
-    if bit_depth == 8:
-        dtype = 'u1'
-    else:
-        if format_tag == WAVE_FORMAT.PCM:
+
+    if format_tag == WAVE_FORMAT.PCM:
+        if 1 <= bit_depth <= 8:
+            dtype = 'u1'  # WAV of 8-bit integer or less are unsigned
+        elif bytes_per_sample in {3, 5, 6, 7}:
+            # No compatible dtype.  Load as raw bytes for reshaping later.
+            dtype = 'V1'
+        elif bit_depth <= 64:
```

Comment:
Would this be more clear as `elif bytes_per_sample in (2,4,8):`?

Reply:
Hmmm

- Whether the data is signed vs unsigned is determined by the bit depth being 8 or less bits. 
- Whether the data corresponds to a native dtype vs needs reshaping is determined by the container byte size.

These aren't necessarily related, either.  A WAV file could store unsigned 5-bit data in 4-byte containers, for example, which ...

Wait, this doesn't work, does it?  If a WAV stores 8-bit data in a 3-byte container, then we need to interpret it as unsigned, because it's <= 8-bit, but reshape it to fit in 4-byte container, since there's no uint24, so it should be reshaped to dtype=uint32?

http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/Docs/riffmci.pdf does say "The size of i is the smallest number of bytes required to contain the specified sample size." so I guess that's not legal (and probably doesn't exist anyway).

and http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html describes a real-world format with 24 bit depth but 4 byte container size which is actually float stored as int.  So we wouldn't want to open that as int.  

So it should only accept files that have a container size that doesn't exceed the bit depth?  And if someone wants to store 24-bit data and read it back mmapped, they can store it as right-shifted int32?  (Again, most of these corner cases will never be seen by anyone.)